File: cs50/pset6/sentimental/vigenere/caesar/caesar.py
import sys

lower_low = ord('a')
lower_high = ord('z')
upper_low = ord('A')
upper_high = ord('Z')


def main():
    if(len(sys.argv) != 2):
        print("Usage: python ceasay.py k")
        exit(1)

    else:
        string = input("plaintext: ")
        print("ciphertext: ", end="")
        for c in string:
            c2 = ord(c)
            if(c2 >= lower_low and c2 <= lower_high):
                print(chr((((c2 + int(sys.argv[1])) - lower_low) % 26) + lower_low), end="")
            elif(c2 >= upper_low and c2 <= upper_high):
                print(chr((((c2 + int(sys.argv[1])) - upper_low) % 26) + upper_low), end="")

            else:
                print(c, end="")

    print()
    return 0


main()

File: cs50/pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>


int main(int argc, string argv[])
{
    const int lower_low = 'a';
    const int lower_high = 'z';
    const int upper_low = 'A';
    const int upper_high = 'Z';

    if (argc != 2)
    {
        printf("Usage: ./caesar k\n");
        return 1;
    }
    else
    {
        string input = get_string("plaintext:  ");
        printf("ciphertext: ");
        for (int i = 0, n = strlen(input); i < n; i++)
        {
            if (input[i] >= lower_low && input[i] <= lower_high)
            {
                char result = (((input[i]+atoi(argv[1])-'a')%26)+'a');
                printf("%c", result);
            }
            else if (input[i] >= upper_low && input[i] <= upper_high)
            {
                char result = (((input[i]+atoi(argv[1])-'A')%26)+'A');
                printf("%c", result);
            }
            else
            {
                printf("%c", input[i]);
            }
        }
        printf("\n");
    }
}